core/src/rigidbody.cpp

```diff
@@ -0,0 +1,289 @@
+    if (m_dynamic) {
+        m_dynamic->setGlobalPose(PxTransform(p));
+        m_dynamic->setGlobalPose(PxTransform(q));
+    } else {
+        m_static->setGlobalPose(PxTransform(p));
+        m_static->setGlobalPose(PxTransform(q));
+    }
+}
+
+void RigidBody::set_position(const vec3& position) {
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    physx::PxVec3 p = physx::PxVec3(p_x, p_y, p_z);
+    if (m_dynamic) {
+        m_dynamic->setGlobalPose(PxTransform(p));
+    } else {
+        m_static->setGlobalPose(PxTransform(p));
+    }
+}
+void RigidBody::set_rotation(const quat& rotation) {
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
+    physx::PxQuat q = physx::PxQuat(r_w, r_x, r_y, r_z);
+    if (m_dynamic) {
+        m_dynamic->setGlobalPose(PxTransform(q));
+    } else {
+        m_static->setGlobalPose(PxTransform(q));
+    }
+}
+
+void RigidBody::set_material(physx::PxMaterial* material) {
+    m_material = material;
+}
+
+void RigidBody::set_physics_and_scene(physx::PxScene* scene, physx::PxPhysics* physics) {
+    m_scene = scene;
+    m_physics = physics;
+}
+
+void RigidBody::create_cube_rigid_dynamic(const vec3& halfSize,
+                                          const float& mass,
+                                          const vec3& position,
+                                          const quat& rotation) {
+    if (m_static) {
+        return;
+    }
+    physx::PxShape* shape = NULL;
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
+    physx::PxVec3 p = physx::PxVec3(p_x, p_y, p_z);
+    physx::PxQuat q = physx::PxQuat(r_w, r_x, r_y, r_z);
+    if (m_material) {
+        shape = m_physics->createShape(physx::PxBoxGeometry(halfSize.x, halfSize.y, halfSize.z), *m_material);
+    } else {
+        m_material = m_physics->createMaterial(0.3f, 0.3f, 0.6f);
+        shape = m_physics->createShape(physx::PxBoxGeometry(halfSize.x, halfSize.y, halfSize.z), *m_material);
+    }
+    m_dynamic = m_physics->createRigidDynamic(PxTransform(p));
+    m_dynamic->setGlobalPose(PxTransform(q));
+    m_dynamic->attachShape(*shape);
+    PxRigidBodyExt::updateMassAndInertia(*m_dynamic, mass);
+    m_scene->addActor(*m_dynamic);
+    shape->release();
+}
+
+void RigidBody::create_cube_rigid_static(const vec3& halfSize, const vec3& position, const quat& rotation) {
+    if (m_dynamic) {
+        return;
+    }
+    physx::PxShape* shape = NULL;
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
+    physx::PxVec3 p = physx::PxVec3(p_x, p_y, p_z);
+    physx::PxQuat q = physx::PxQuat(r_w, r_x, r_y, r_z);
```

Comment:
You are repeating this expression too often, create a helper to convert `vec3` to `PxVec3`, same with any other types you constantly need.